deepinv/training/trainer.py

```diff
@@ -33,69 +33,11 @@ class Trainer:
 
     where ``.pth.tar`` file contains a dictionary with the keys: ``epoch`` current epoch, ``state_dict`` the state
     dictionary of the model, ``loss`` the loss history, ``optimizer`` the state dictionary of the optimizer,
-    and ``eval_metrics` the evaluation metrics history.
-
-    The class provides a flexible training loop that can be customized by the user. In particular, the user can
-    rewrite the :meth:`deepinv.Trainer.compute_loss` method to define their custom training step without having
-    to write all the training code from scratch:
-
-
-    ::
-
-        class CustomTrainer(Trainer):
-            def compute_loss(self, physics, x, y, train=True, epoch: int = None):
-                logs = {}
-
-                self.optimizer.zero_grad() # Zero the gradients
-
-                # Evaluate reconstruction network
-                x_net = self.model_inference(y=y, physics=physics)
-
-                # Compute the losses
-                loss_total = 0
-                for k, l in enumerate(self.losses):
-                    loss = l(x=x, x_net=x_net, y=y, physics=physics, model=self.model, epoch=epoch)
-                    loss_total += loss.mean()
-
-                metric = self.logs_total_loss_train if train else self.logs_total_loss_eval
-                metric.update(loss_total.item())
-                logs[f"TotalLoss"] = metric.avg
-
-                if train:
-                    loss_total.backward()  # Backward the total loss
-                    self.optimizer.step() # Optimizer step
-
-                return x_net, logs
-
-
-    If the user wants to change the way the metrics are computed, they can rewrite the
-    :meth:`deepinv.Trainer.compute_metrics` method.
-    The user can also change the way samples are generated by overriding either the :meth:`deepinv.Trainer.get_samples_online` method or the :meth:`deepinv.Trainer.get_samples_offline` method,
-    e.g. to change the physics parameters on-the-fly with parameters from the dataset.
+    and ``eval_metrics`` the evaluation metrics history.
```

Comment:
I don't know how to see both, but I will add a link to the user guide.

I thought that it would be nice to declutter a bit the class description, and move the explanations to the user guide

Reply:
I've added a link to the user guide